### ISSUE TITLE
2.7 InvokableFactory options from AbstractPluginManager affect multiple unshared instantiations

### ISSUE THREAD

**Issue by AndrolGenhald** [zendframework/zend-servicemanager#126]
I ran into this when using a NotEmpty validator, if I specify the options for one of them, all of them get those options rather than the defaults.

```
public function getInputFilterSpecification()
{
	return [
		'elementA' => [
			'required' => true,
			'validators' => [
				[
					'name' => 'notEmpty',
					'options' => [
						'type' => NotEmpty::NULL,
					],
				],
			],
		],
		'elementB' => [
			'required' => true,
			//Injected validator will have type NotEmpty::NULL
		],
	];
}
```

**Comment by AndrolGenhald** [comment]
I added a test and tried to fix it by having AbstractPluginManager call setCreationOptions with an empty array, but that causes testMutableMethodNeverCalledWithoutCreationOptions to fail.
AndrolGenhald/zend-servicemanager@c56f494d0948697ac2867d0b2aa4eb877a395c37

**Comment by weierophinney** [comment]
Fixed with #127.
